src/main/java/com/actiontech/dble/services/rwsplit/RWSplitService.java

```diff
@@ -237,12 +241,31 @@ private void handleComQuery(byte[] data) {
             }
             executeSql = sql;
             executeSqlBytes = data;
+            if (blacklistCheck(sql, userConfig.getBlacklist())) return;
             queryHandler.query(sql);
         } catch (UnsupportedEncodingException e) {
             writeErrMessage(ErrorCode.ER_UNKNOWN_COM_ERROR, e.getMessage());
         }
     }
 
+    private boolean blacklistCheck(String sql, WallProvider blackList) {
+        if (Objects.isNull(blackList)) return false;
+        WallCheckResult result = blackList.check(sql);
```

Comment:
whether support for 'BEGIN' syntax is considered?

Reply:
in rwsplit, the blacklist parse unsupported syntax is still issued after execution